pkg/operation/botanist/secrets.go

```
@@ -441,6 +452,16 @@ func (b *Botanist) SyncShootCredentialsToGarden(ctx context.Context) error {
 		},
 	}
 
+	// ssh-keypair.old secret will be synced to the Garden cluster if it is present in shoot namespace in seed.
+	oldSecret := &corev1.Secret{}
+	if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, v1beta1constants.SecretNameOldSSHKeyPair), oldSecret); err == nil {
```

Do not silently ignore the error here

---

Here `err` will not be nil when the shoot is newly created. For newly created shoot there will be no `ssh-keypair.old` secret. Hence I am only considering the case where `ssh-keypair.old` secret is present and only in that case secret will be synced to garden cluster.

---

Sure, but please handle "IsNotFound" errors and all other errors accordingly. Just silently ignoring errors is not a good practice.

---

Addressed in [commit](https://github.com/gardener/gardener/pull/5388/commits/328decb0af508ddc927789babd8b18b9d8d7b66a)